browser/ui/views/brave_icon_with_badge_image_source.h

```diff
@@ -0,0 +1,56 @@
+/* Copyright (c) 2020 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#ifndef BRAVE_BROWSER_UI_VIEWS_BRAVE_ICON_WITH_BADGE_IMAGE_SOURCE_H_
+#define BRAVE_BROWSER_UI_VIEWS_BRAVE_ICON_WITH_BADGE_IMAGE_SOURCE_H_
+
+#include "chrome/browser/ui/extensions/icon_with_badge_image_source.h"
+
+namespace gfx {
+class Canvas;
+class Rect;
+}  // namespace gfx
+
+namespace brave {
+
+// The purpose of this subclass is to:
+// - Paint the BraveAction badge in a custom location and with a different size
+//   to regular BrowserAction extensions.
+class BraveIconWithBadgeImageSource : public IconWithBadgeImageSource {
```

Comment:
For the change of this file, it would be good to have @petemill (the original author) or @nullhook (who recently changed this file) to take a look.